src/Text/Pandoc/Writers/DocBook.hs

```diff
@@ -464,6 +465,14 @@ idAndRole (id',cls,_) = ident <> role
     ident = [("id", id') | not (T.null id')]
     role  = [("role", T.unwords cls) | not (null cls)]
 
+-- Used in blockToDocBook for Header (section) to create or extend
+-- the role attribute with candidate class tokens
+enrichRole :: [(Text, Text)] -> [Text] -> [(Text, Text)]
+enrichRole mattrs cls = [("role",rolevals) | rolevals /= ""]<>(filter (\x -> (fst x) /= "role") mattrs)
+  where
+    rolevals = T.unwords((filter (`elem` cand) cls)<>(maybeToList(lookup "role" mattrs)))
+    cand = ["unnumbered"]
```

Comment:
Nice, thanks! This is good to merge from my POV.

Your comments on readability made me wonder, and I ended up writing an (untested!!) version trying to strike a balance between readability and and conciseness. Not sure if I succeeded, but here we go:

``` haskell
enrichRole :: [(Text, Text)] -> [Text] -> [(Text, Text)]
enrichRole mattrs cls = [("role", T.unword roles) | not (null roles)] <> nonRole
  where
    (roleAttr, nonRole) = partition (\(key, _v) -> k == "role") mattrs
    roles = nub $ ["unnumbered" | "unnumbered" `elem` cls] <> map snd roleAttr
```

Reply:
@tarleb Thank you for the suggestion! I'm fine either way, although having `cand` separate was something I hoped to keep, as I can envision there are more class tokens that would make sense to add as role tokens in DocBook, and adding to that list is (perhaps) less intimidating to future contributors. Let me know if you would prefer it changed, and if not, if there's more to do before merge (rebase on current master?)

Reply:
Agreed, keeping the list of "role classes" separate seems sensible.

I just noticed that some lines break the 80 chars limit, we generally want to keep lines below that length. Once that's fixed we can squash-merge everything into a single commit; no need for a rebase.

Reply:
@tarleb Sorry, didn't get to the 80char fix before it merged. Will keep it in mind in the future.

Reply:
No problem at all. Thanks again, looking forward to the next one!